### PR TITLE
fix payload encoding and code refactor

### PR DESCRIPTION
**Pull request by xavluiz**
* fixes 2 main issues
  - encodes the payload to fix: https://www.notion.so/softwaredotcom/Sublime-Text-console-error-21b7bf55b45e4eb599c383a70df04eb3
  -  adds the missing argument to the create anon function as this logic hasn't been working
 * code clean up

### CODE REVIEW

File: Software.py [hunk 1]
```diff
@@ -282,7 +281,7 @@ def on_modified_async(self, view):
         global PROJECT_DIR
         # get active data will create the file info if it doesn't exist
         active_data = PluginData.get_active_data(view)
-        if active_data is None:
+        if active_data is None or active_data.project is None:
```
Comment (brettmstevens7) [inline Software.py:284]: Will this exclude tracking files edited outside of projects?

File: lib/SoftwareUserStatus.py [hunk 2]
```diff
@@ -100,6 +101,33 @@ def refetchUserStatusLazily(tryCountUntilFoundUser):
 
         updateSessionSummaryFromServer(True)
 
+def authCompletionHandler(user):
```
Comment (brettmstevens7) [inline lib/SoftwareUserStatus.py:104]: Does this get called anywhere?

File: Software.py (hunk 1 above)
Comment (xavluiz) [reply Software.py:284]: We do our best to create the Untitled project but I noticed during testing that we'll occasionally get a NoneType project. This is to just prevent error chains from causing bigger issues. And based on testing, we'll create the project the next keystroke or two.

File: lib/SoftwareUserStatus.py (hunk 2 above)
Comment (xavluiz) [reply lib/SoftwareUserStatus.py:104]: oh that. that was to prepare for the websocket logic. I had websocket-client and that kept failing due to not finding another module. I decided to take out the websocket logic until I figure that out

Comment (xavluiz) [reply lib/SoftwareUserStatus.py]: i removed that method for now

**Review by brettmstevens7 — APPROVED** [review]

**Pull request merged**